libflux/src/libstd/lib.rs

```diff
@@ -19,6 +24,50 @@ pub fn fresher() -> Fresher {
     flatbuffers::get_root::<fb::Fresher>(buf).into()
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn flux_semantic_analyze(
+    src_ptr: *const c_char,
+    flux_buf: *mut flux_buffer_t,
+) -> *mut flux_error_t {
+    let buf = CStr::from_ptr(src_ptr).to_bytes(); // Unsafe
+    let s = String::from_utf8(buf.to_vec()).unwrap();
+    match analyze(s.as_str()) {
+        Ok(vec) => {
+            let flux_buf = &mut *flux_buf;
+            flux_buf.data = vec.as_ptr();
+            flux_buf.len = vec.len();
+            std::mem::forget(vec);
+            std::ptr::null_mut()
+        }
+        Err(err) => {
+            let errh = flux::ErrorHandle { err: Box::new(err) };
+            return Box::into_raw(Box::new(errh)) as *mut flux_error_t;
+        }
+    }
+}
+
+fn analyze(src: &str) -> Result<Vec<u8>, flux::Error> {
```

Comment:
Should we make this a public method? So that it can be considered the entry point into performing type inference from the Rust API? 

Also I think our terminology is a bit inconsistent at this point. Should this method be called analyze or something else? Previously analyze referred only to the step of converting ast to semantic graph but without the type inference bits.  This method includes parsing, converting ast -> semantic and running type inference. I actually like the name analyze but maybe we should rename the `semantic/analyze` package to `convert` or something different so to not confuse the two.